File: haskell/sum-of-multiples/src/SumOfMultiples.hs
module SumOfMultiples (sumOfMultiples) where

sumOfMultiples :: [Integer] -> Integer -> Integer
sumOfMultiples factors limit = sum $ filter modEach [1..limit - 1]
    where modEach x = any ((0 ==) . (x `mod`)) $ filter (0 /=) factors

File: clojure/sum-of-multiples/src/sum_of_multiples.clj
(ns sum-of-multiples)

(defn multiple? [multiples n]
  (some true? (map #(= 0 (mod n %)) multiples)))

(defn sum-of-multiples [multiples n]
  (apply + (filter #(multiple? multiples %) (range 1 n))))
